src/scopes.js

```diff
@@ -56,3 +60,35 @@ exports.scopeMatch = function(scopePatterns, scopesets) {
     });
   });
 };
+
+/**
+ * Finds scope intersections between two scope sets.
+ */
+exports.scopeIntersection = (scopeset1, scopeset2) => {
+  validateScopePatterns(scopeset2);
+  validateScopePatterns(scopeset1);
+
+  const junctions = [];
+  const scope1 = scopeset2.filter(s => s);
+  const scope2 = scopeset1.filter(s => s);
+
+  scope2.forEach((s2) => {
+    scope1.forEach((s1) => {
+      const star = '*';
+      const s1IndexOfStar = s1.indexOf(star);
+      const s2IndexOfStar = s2.indexOf(star);
```

Comment:
This is alarming, what is going on here? `*` is not interesting unless it's at the end of the scope.